src/Compiler/pars.fsy

```diff
@@ -4891,9 +4891,28 @@ atomicExpr:
         parseState.LexBuffer.CheckLanguageFeatureAndRecover LanguageFeature.AccessorFunctionShorthand (unionRanges mUnderscore  mDot )
         reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnderScoreDotLambdaNonAtomic())
         let expr = $3
-        let trivia: SynExprDotLambdaTrivia = { UnderscoreRange = mUnderscore ; DotRange = mDot  }
-        SynExpr.DotLambda(expr, unionRanges mUnderscore  expr.Range, trivia), false }    
-        
+        let trivia: SynExprDotLambdaTrivia = { UnderscoreRange = mUnderscore ; DotRange = Some mDot  }
+        SynExpr.DotLambda(expr, unionRanges mUnderscore  expr.Range, trivia), false }
+
+  | UNDERSCORE DOT recover %prec dot_lambda
+      { let mUnderscore = rhs parseState 1
+        let mDot = rhs parseState 2
+        parseState.LexBuffer.CheckLanguageFeatureAndRecover LanguageFeature.AccessorFunctionShorthand (unionRanges mUnderscore mDot)
+        let expr = arbExpr ("dotLambda1", mDot.EndRange)
+        let trivia: SynExprDotLambdaTrivia = { UnderscoreRange = mUnderscore ; DotRange = Some mDot  }
```

Comment:
I would expect a parser error here, what am I missing?

Reply:
`recover` will report an error in face of a wrong token, so this is fine in the most cases. However, a special case is needed for `EOF` and it isn't handled in this rule. Thanks for catching!